hail/src/main/scala/is/hail/expr/ir/BaseIR.scala

```diff
@@ -13,10 +13,10 @@ abstract class BaseIR {
 
   def children: Iterable[BaseIR] = childrenSeq
 
-  protected def copy(newChildren: IndexedSeq[BaseIR]): BaseIR
+  protected def withNewChildren(newChildren: IndexedSeq[BaseIR]): BaseIR
```

Comment:
`with*` names suggest some sort of temporary context. Would be nice to honour that convention and rename this to `setNewChildren` or something.

Reply:
Set sound like it mutates. How about `copyWithNewChildren`?